exploration/0477-quoted-literals.md

```diff
@@ -0,0 +1,268 @@
+
+  > ```
+  > match ($count :choice}
+  > when |<10| {A handful.}
+  > when |11..19| {Umpteen.}
+  > when * {Lots.}
+  >
+  > match {$arbitraryString}
+  > when |can't resolve| {Can't resolve!}
+  > when |11'233.44| {Locale formatted number}
+  > when |New York| {A multi-word proper name}
+  > when * {Imagine more...}
+  > ```
+
+- Message authors may want to protect untranslatable strings:
+
+  > ```
+  > {Visit {|http://www.example.com| @translate=false}.}
+  > ```
+  >
+  > See [design proposal 0002](https://github.com/unicode-org/message-format-wg/blob/main/exploration/0002-expression-attributes.md).
+
+- Message authors may want to decorate substrings as being written in a particular language,
+  different from the message's language,
+  for the purpose of accessibility, text-to-speech, and semantic correctness.
+
+  > ```
+  > {The official native name of the Republic of Poland is {|Rzeczpospolita Polska| @lang=pl}.}
+  > ```
+  >
+  > See [design proposal 0002](https://github.com/unicode-org/message-format-wg/blob/main/exploration/0002-expression-attributes.md).
+
+- Developers may want to embed messages with quoted literals in code written in another programming language
+  which uses single or double quotes to delimit strings.
+
+  > ```js
+  > let message = new MessageFormat("en", "{A message with {|a literal|}.}");
+  > ```
+
+- Developers and localization engineers may want to embed messages with quoted literals in a container format, such as JSON.
+
+  > ```json
+  > {
+  >   "msg": "{A message with {|a literal|}.}"
+  > }
+  > ```
+
+## Requirements
+
+_What properties does the solution have to manifest to enable the use-cases above?_
+
+- **[r1; high priority]** Minimize the need to escape characters inside literals.
+  In particular, choose a delimiter that isn't frequently used in translation content.
+  Having to escape characters inside literals is inconvenient and error-prone when done by hand,
+  and it also introduces the backslash into the message, `\`,
+  which is the escape introducer.
+  The backslash then needs to be escaped too,
+  when the message is embedded in code or containers.
+  (This is how some syntaxes produce the gnarly `\\\`.)
+
+- **[r2; high priority]** Minimize the need to escape characters when embedding messages in code or containers.
+  In particular, choose a delimiter that isn't frequently used as a string delimiter in programming languages and container formats.
+  However, note that many programming languages also provide alternative ways of delimiting strings, e.g. _raw strings_ or triple-quoted literals.
+
+- **[r3; medium priority]** Minimize the incentive to avoid escaping by changing messages
+  (e.g. rephrasing content, using typographic apostrophes, or switching outer delimiters).
+
+- **[r4; medium priority]** Don't surprise users with syntax that's too exotic.
+  We expect quoted literals to be rare,
+  which means fewer opportunities to get used to their syntax and remember it.
```

Comment:
I think the priority of this requirement should be ranked higher. I also realise that I ought to look into the data I have available to establish how rare the need for quoting will be, as that might be better than relying just on our expectations.

```suggestion
- **[r4; high priority]** Don't surprise users with syntax that's too exotic.
  We expect quoted literals to be rare,
  which means fewer opportunities to get used to their syntax and remember it.
```

Reply:
This (and later comments) about differing priorities is why I suggested a table. We might not agree about the specific priorities of this or that item, but it's hard to compare options when they are discrete bits of text spread over many paragraphs. It's easier to "rescore" when everything is in one place.

Reply:
I'll change this to `medium/high` to capture the difference in priority scores.